src/main/java/macrobase/ingest/SQLLoader.java

```diff
@@ -88,6 +93,7 @@ public RowSet getRows(String baseQuery,
                           List<RowSetResource.RowSetRequest.RowRequestPair> preds,
                           int limit,
                           int offset) throws SQLException {
+        // TODO do we need to handle time column here?
```

Comment:
Yeah, `baseQuery` is a hack. As an intermediate option, you could check for `ORDER BY` and error out.